src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSourceProvider.cs

```diff
@@ -335,64 +360,102 @@ private SuggestedActionSet OrganizeRefactorings(Workspace workspace, CodeRefacto
                 return new SuggestedActionSet(refactoringSuggestedActions.ToImmutable(), SuggestedActionSetPriority.Low);
             }
 
-            public async Task<bool> HasSuggestedActionsAsync(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
+            public async Task<bool> HasSuggestedActionsAsync(
+                ISuggestedActionCategorySet requestedActionCategories,
+                SnapshotSpan range,
+                CancellationToken cancellationToken)
             {
-                var view = _textView;
```

Comment:
it is not our code. since we clear up any field at dispose, we need to make sure any Async method we expose to others, we need to make sure we hold onto information we need. basically, those method should never use field.

things like GetSuggestedAction that runs on UI thread doesnt need to do so, since it is synchronous and run on UI thread. so field can't be cleared. but things that will be called from background thread and async, those code path should never use field. and that is why it explicitly hold onto those fields.